packager/bindlepublisher.go

```diff
@@ -0,0 +1,225 @@
+package packager
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/deislabs/go-bindle/client"
+	"github.com/deislabs/go-bindle/keyring"
+	"github.com/deislabs/go-bindle/types"
+	"github.com/pelletier/go-toml"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+
+	"github.com/suborbital/subo/project"
+	"github.com/suborbital/subo/subo/util"
+)
+
+const (
+	bindlePublishJobType = "bindle"
+	suboAuthor           = "Subo <subo@suborbital.dev>"
+)
+
+type BindlePublishJob struct{}
+
+type parcelWrapper struct {
+	parcel types.Parcel
+	data   []byte
+}
+
+// NewBindlePublishJob returns a new PublishJob for Bindle.
+func NewBindlePublishJob() PublishJob {
+	b := &BindlePublishJob{}
+
+	return b
+}
+
+// Type returns the publish job's type.
+func (b *BindlePublishJob) Type() string {
+	return bindlePublishJobType
+}
+
+// Publish publishes the application.
+func (b *BindlePublishJob) Publish(log util.FriendlyLogger, ctx *project.Context) error {
+	if ctx.Directive == nil {
+		return errors.New("🚫 cannot push without Directive.yaml file")
+	}
+
+	log.LogStart(fmt.Sprintf("pushing %s@%s", ctx.Directive.Identifier, ctx.Directive.AppVersion))
+
+	invoice := &types.Invoice{
+		BindleVersion: "1.0.0",
+		Bindle: types.BindleSpec{
+			Name:    ctx.Directive.Identifier,
+			Version: strings.TrimPrefix(ctx.Directive.AppVersion, "v"),
+			Authors: []string{
+				suboAuthor,
+			},
+		},
+		Parcel: []types.Parcel{},
+	}
+
+	parcelsBySHA := map[string]parcelWrapper{}
```

Comment:
I'm afraid it wouldn't be consistent, since the convention in our repos thus far has been to only use `make` when the size of the slice is known beforehand.